includes/class-amp-story-post-type.php

```diff
@@ -986,184 +986,21 @@ public static function get_fonts() {
-			],
-			[
-				'name'      => 'Noto Sans',
-				'gfont'     => 'Noto Sans:400,700',
-				'fallbacks' => [ 'sans-serif' ],
-			],
-			[
-				'name'      => 'Open Sans',
-				'gfont'     => 'Open Sans:400,700',
-				'fallbacks' => [ 'sans-serif' ],
-			],
-			[
-				'name'      => 'Open Sans Condensed',
-				'gfont'     => 'Open Sans Condensed:400,700',
-				'fallbacks' => [ 'sans-serif' ],
-			],
-			[
-				'name'      => 'Oswald',
-				'gfont'     => 'Oswald:400,700',
-				'fallbacks' => [ 'sans-serif' ],
-			],
-			[
-				'name'      => 'Palatino',
-				'fallbacks' => [ 'Palatino Linotype', 'Palatino LT STD', 'Book Antiqua', 'Georgia', 'serif' ],
-			],
-			[
-				'name'      => 'Papyrus',
-				'fallbacks' => [ 'fantasy' ],
-			],
-			[
-				'name'      => 'Playfair Display',
-				'gfont'     => 'Playfair Display:400,700',
-				'fallbacks' => [ 'serif' ],
-			],
-			[
-				'name'      => 'PT Sans',
-				'gfont'     => 'PT Sans:400,700',
-				'fallbacks' => [ 'sans-serif' ],
-			],
-			[
-				'name'      => 'PT Sans Narrow',
-				'gfont'     => 'PT Sans Narrow:400,700',
-				'fallbacks' => [ 'sans-serif' ],
-			],
-			[
-				'name'      => 'PT Serif',
-				'gfont'     => 'PT Serif:400,700',
-				'fallbacks' => [ 'serif' ],
-			],
-			[
-				'name'      => 'Raleway',
-				'gfont'     => 'Raleway:400,700',
-				'fallbacks' => [ 'sans-serif' ],
-			],
-			[
-				'name'      => 'Roboto',
-				'gfont'     => 'Roboto:400,700',
-				'fallbacks' => [ 'sans-serif' ],
-			],
-			[
-				'name'      => 'Roboto Condensed',
-				'gfont'     => 'Roboto Condensed:400,700',
-				'fallbacks' => [ 'sans-serif' ],
-			],
-			[
-				'name'      => 'Roboto Slab',
-				'gfont'     => 'Roboto Slab:400,700',
-				'fallbacks' => [ 'serif' ],
-			],
-			[
-				'name'      => 'Slabo 27px',
-				'gfont'     => 'Slabo 27px:400,700',
-				'fallbacks' => [ 'serif' ],
-			],
-			[
-				'name'      => 'Source Sans Pro',
-				'gfont'     => 'Source Sans Pro:400,700',
-				'fallbacks' => [ 'sans-serif' ],
-			],
-			[
-				'name'      => 'Tahoma',
-				'fallbacks' => [ 'Verdana', 'Segoe', 'sans-serif' ],
-			],
-			[
-				'name'      => 'Times New Roman',
-				'fallbacks' => [ 'Times New Roman', 'Times', 'Baskerville', 'Georgia', 'serif' ],
-			],
-			[
-				'name'      => 'Trebuchet MS',
-				'fallbacks' => [ 'Lucida Grande', 'Lucida Sans Unicode', 'Lucida Sans', 'Tahoma', 'sans-serif' ],
-			],
-			[
-				'name'      => 'Ubuntu',
-				'gfont'     => 'Ubuntu:400,700',
-				'fallbacks' => [ 'sans-serif' ],
-			],
-			[
-				'name'      => 'Verdana',
-				'fallbacks' => [ 'Geneva', 'sans-serif' ],
-			],
-		];
+		$file_content = file_get_contents( __DIR__ . '/data/fonts.json' );  // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+		$google_fonts = json_decode( $file_content, true );
+		$fonts        = [];
+		foreach ( $google_fonts['items'] as $font ) {
+			$gfont    = '';
+			$variants = array_diff( $font['variants'], [ 'regular' ] );
```

Comment:
I think `regular` should not be excluded, but instead replaced with `400`.

Example:

When I want to download the Ubuntu font from Google Fonts in regular and bold variants, I get a URL like this:

```
https://fonts.googleapis.com/css?family=Ubuntu:400,700&display=swap
```

Reply:
Can you explain this?

Reply:
Taking the `Ubuntu` font here as an example:

In the `develop` branch, the `gfont` entry for that font is `Ubuntu:400,700`.

In `fonts.json`, the following variants are listed for `Ubuntu`:

```
[
	"300",
	"300italic",
	"regular",
	"italic",
	"500",
	"500italic",
	"700",
	"700italic"
]
```

In this PR, this is turnt into `Ubuntu:300,300italic,italic,500,500italic,700,700italic`, whereas the correct entry ([as output by the Google Fonts UI](https://fonts.google.com/specimen/Ubuntu?selection.family=Ubuntu:300,300i,400,400i,500,500i,700,700i)) would be `Ubuntu:300,300i,400,400i,500,500i,700,700i`.

So `regular` is turned into `400`, and `300italic` is turned into `300i`, `700italic` into `700i`, etc.

Given that we only support regular, italic, and bold in the stories editor UI, I would strip out everything else, so that we get `Ubuntu:400,400i,700,700i` in the end.

Hope that makes sense.

If not, happy to amend a test case to demonstrate this.

Reply:
Regular should excluded here. By loading any font, it will just automatically load the regular. Take this for example https://fonts.googleapis.com/css?family=Ubuntu . Just loads the regular (400). So it is not required. Which is why it was ignored.  

Good spot, `700italic` -> `700i`. But this is a WIP, so I hadn't tests the fonts loading yet.

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/3110/commits/3a16fadde9a3858d339de32d187df9501063b69d